apps/client/src/services/utils.ts

```diff
@@ -124,8 +124,31 @@ function formatDateISO(date: Date) {
     return `${date.getFullYear()}-${padNum(date.getMonth() + 1)}-${padNum(date.getDate())}`;
 }
 
-function formatDateTime(date: Date) {
-    return `${formatDate(date)} ${formatTime(date)}`;
+
+// old version
+// function formatDateTime(date: Date) {
+//     return `${formatDate(date)} ${formatTime(date)}`;
+// }
+
+// new version 
+export function formatDateTime(date: Date, userSuppliedFormat?: string): string { 
+    const DEFAULT_FORMAT = 'YYYY-MM-DD HH:mm'; 
+    let formatToUse = DEFAULT_FORMAT;
+
+    if (userSuppliedFormat && typeof userSuppliedFormat === 'string' && userSuppliedFormat.trim() !== "") {
+        formatToUse = userSuppliedFormat.trim();
+    }
```

Comment:
I think that block can be simplified by using a ternary operator into

```
const formatToUse = (typeof userSuppliedFormat === 'string' && userSuppliedFormat.trim() !== "") 
    ? userSuppliedFormat.trim()
    : DEFAULT_FORMAT;
```

That first `userSuppliedFormat` "falsey" check was redundant anyways, if I am not mistaken:
* `typeof` already checks if it is a string (so rules out any `undefined`, `null` values)
* `userSuppliedFormat.trim() !== ""` filters out "empty" strings as well